lib/common/flow.js

```diff
@@ -34,6 +34,16 @@ Flow.prototype = {
   },
 
   warn: function(warn){
+    if (this.options.ignoreWarnings &&
+        !warn.fatal &&
+        warn.file &&
+        this.options.ignoreWarnings.some(function(pattern){
+      return minimatch(warn.file, pattern);
```

Comment:
Also create pattern via `new Minimatch(..)` and further `pattern.match()` check may perform better then `minimatch(filename, pattern)` (see example https://github.com/basisjs/basisjs-tools-build/blob/master/lib/common/plugins.js#L22)